frontend/src/components/tracing/tracing.tsx

```diff
@@ -101,24 +107,24 @@ export const Tracing: React.FC = () => {
 };
 
 const LazyVega = React.lazy(() =>
-  import("react-vega").then((m) => ({ default: m.Vega })),
+  import("react-vega").then((m) => ({ default: m.VegaLite })),
 );
 interface ChartProps {
   className?: string;
-  width: number;
   height: number;
   vegaSpec: VisualizationSpec;
   signalListeners: SignalListeners;
   theme: ResolvedTheme;
 }
 
 const Chart: React.FC<ChartProps> = (props: ChartProps) => {
+  const { ref, width = 300 } = useResizeObserver<HTMLDivElement>();
```

Comment:
@Light2Dark , let me know what you think about this. happy to remove

Reply:
i'm okay with this :), cool solution